hal/src/dma/controller.rs

```diff
@@ -0,0 +1,270 @@
+//! DMA controller (DMAC)
+
+use vcell::VolatileCell;
+
+/// (DMAC_BTCTRL) Source Address Increment Enable
+pub(super) const BTCTRL_SRCINC: u16 = 1 << 10;
```

Comment:
It'd be nice to get rid of all of these consts and replace them with things from the SVD, or failing that proper `bitfield` types.

A lot of them are for DMAC descriptor fields.

Reply:
https://github.com/sajattack/atsamd/blob/9102fe969e6d85639ec70faf2c544fe4691ffc2c/hal/src/dma.rs#L183